### PR TITLE
INTMDB-232: Fix user agent version

### PR DESCRIPTION
**Pull request by MihaiBojin**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s):

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the Terraform contribution guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Further comments
Confirmed UA is now reported correctly (Splunk):

### Repro

1. Create a file called `main.tf`

```hcl
terraform {
  required_providers {
    mongodbatlas = {
      source = "localhost/mongodb/mongodbatlas"
      version = "1.0.0"
    }
  }
}

provider "mongodbatlas" {
  public_key = "..."
  private_key  = "..."
}


resource "mongodbatlas_project" "p0" {
  name   = "Project 0"
  org_id = "$ORG_ID"
}
```

2. Compile the provider and install it locally:
```shell
mkdir -p  ~/.terraform.d/plugins/localhost/mongodb/mongodbatlas/1.0.0/darwin_amd64
make build && mv ~/go/bin/terraform-provider-mongodbatlas ~/.terraform.d/plugins/localhost/mongodb/mongodbatlas/1.0.0/darwin_amd64/
```

3. Init the config and import a project (making a call to the API)
```shell
rm -rf .terraform .terraform.lock.hcl && terraform init &&  terraform import mongodbatlas_project.p0 $ORG_ID
```

### Reported UA
Screenshot 2021-08-05 at 00 29 11

### CODE REVIEW

**Review by themantissa — APPROVED** [review]
LGTM!  Thank you so much @MihaiBojin

File: GNUmakefile [hunk 1]
```diff
@@ -42,7 +42,7 @@ websitefmtcheck:
 lint:
 	@echo "==> Checking source code against linters..."
 	# https://github.com/golangci/golangci-lint/issues/337 fixing error
-	golangci-lint run ./$(PKG_NAME) -v --deadline=30m
+	bin/golangci-lint run ./$(PKG_NAME) -v --deadline=30m
```
Comment (MihaiBojin) [inline GNUmakefile:45]: drive-by: golangci-lint is installed under `bin/` - there's no point relying on the tool being available in the system path. cc @gssbzn in case I'm missing something

File: mongodbatlas/config.go [hunk 2]
```diff
@@ -5,8 +5,9 @@ import (
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/logging"
+	"github.com/mongodb/terraform-provider-mongodbatlas/version"
 
-	digest "github.com/mongodb-forks/digest"
+	"github.com/mongodb-forks/digest"
```
Comment (MihaiBojin) [inline mongodbatlas/config.go:10]: drive-by fix - unnecessary aliasing

File: version/version.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package mongodbatlas
+package version
```
Comment (MihaiBojin) [inline version/version.go:1]: moved under root / following the same directory structure as the AWS provider

File: .goreleaser.yml [hunk 4]
```diff
@@ -14,7 +14,7 @@ builds:
   flags:
     - -trimpath
   ldflags:
-    - -s -w -X mongodbatlas/version.ProviderVersion={{.Version}}
+    - -s -w -X 'github.com/mongodb/terraform-provider-mongodbatlas/version.ProviderVersion={{.Version}}'
```
Comment (MihaiBojin) [inline .goreleaser.yml:17]: ProviderVersion was not being replaced, without the full repo path being specified

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: GNUmakefile (hunk 1 above)
Comment (gssbzn) [reply GNUmakefile:45]: line 16 includes the bin folder in the `PATH` (`export PATH := ./bin:$(PATH)`) for convenience, no need to revert this just curious if that no longer works?

File: mongodbatlas/config.go [hunk 5]
```diff
@@ -38,7 +39,7 @@ func (c *Config) NewClient(ctx context.Context) (interface{}, diag.Diagnostics)
 
 	client.Transport = logging.NewTransport("MongoDB Atlas", transport)
 
-	optsAtlas := []matlasClient.ClientOpt{matlasClient.SetUserAgent("terraform-provider-mongodbatlas/" + ProviderVersion)}
+	optsAtlas := []matlasClient.ClientOpt{matlasClient.SetUserAgent("terraform-provider-mongodbatlas/" + version.ProviderVersion)}
```
Comment (gssbzn) [inline mongodbatlas/config.go:42]: [q] [this](https://github.com/mongodb/mongocli/blob/6ac1f831b2a7d4ae60f4a7d025e4ab3c8da50d79/internal/store/store.go#L35) is the user agent we do for the cli

```go
var userAgent = fmt.Sprintf("%s/%s (%s;%s)", config.ToolName, version.Version, runtime.GOOS, runtime.GOARCH)
```

should we make something similar, ie: include os and arch?

**Review by gssbzn — APPROVED** [review]
LGTM questions and clarifications are not blocking 
Thanks fro fixing this

File: GNUmakefile (hunk 1 above)
Comment (MihaiBojin) [reply GNUmakefile:45]: Didn't work for me on Mac, but it worked in GH actions

File: mongodbatlas/config.go (hunk 5 above)
Comment (MihaiBojin) [reply mongodbatlas/config.go:42]: Yeah that makes sense.  Will update later, but if you want to make the change directly on the PR it would be even better, since I won't be in front of a computer for the next few hours.  Thanks!

Comment (MihaiBojin) [reply mongodbatlas/config.go:42]: On second thought @gssbzn, if we made this change, the UA would be reported as:
```
 userAgent: terraform-provider-mongodbatlas/0.9.1-29-g13eb1636 (darwin;amd64) go-mongodbatlas/0.10.1 (darwin;amd64) 
```

(since we add the OS/ARCH via the go client)

Comment (MihaiBojin) [reply mongodbatlas/config.go:42]: I'm going to merge the PR as is / I don't think duplicating that info makes sense.

Comment (gssbzn) [reply mongodbatlas/config.go:42]: totally forgot we were doing that in the client, merge away
